src/cling/formatter.cr

```diff
@@ -164,5 +171,22 @@ module Cling
       return unless footer = command.footer
       io << footer << "\n\n"
     end
+
+    private def print_parent_command_usage(command : Command, io : IO) : Nil
+      current_command = command
+
+      commands = Array(String).new
+
+      loop do
+        break unless parent = current_command.parent
+
+        current_command = parent
+        commands << current_command.name
+      end
+
+      commands.reverse_each do |command_name|
+        io << command_name << " "
+      end
```

Comment:
I think there's a better way we can handle this: rather than iterating over the commands then iterating over the array, we can recursively call the method for every command that has a parent, then write the command name to the IO.

Reply:
The issue is that you get the command names from right to left. If you write the command name to the IO the order will be wrong.

Reply:
How would that work? I can't think of the implementation right now 🤔